src/main/java/de/unistuttgart/overworldbackend/service/MinigameTaskStatisticService.java

```diff
@@ -72,59 +72,33 @@ public MinigameSuccessRateStatistic getSuccessRateOfMinigame(final UUID minigame
     /**
      * Returns the highscore distribution of a minigame task
      *
+     * For every score between 0 and 100 there is a MinigameScoreHit object with the score and the amount of players that hit that score.
+     * If a score is not hit by any player, it is not included in the list except for 0 and 100.
+     *
      * @param minigameTaskId id of the minigame task to get the statistic for
-     * @param highscorePercentages list of percentages to get the highscore distribution for, e.g. [0, 25, 50, 75, 100]
-     * @throws IllegalArgumentException if highscorePercentages is not sorted, has not as first value 0, at last value 100 or contains values < 0 or > 100
      * @return a minigame highscore distribution for the given minigame task
      */
-    public List<MinigameHighscoreDistribution> getPlayerHighscoreDistributions(
-        final UUID minigameTaskId,
-        final Optional<List<Integer>> highscorePercentages
-    ) {
-        final List<Integer> highscorePercentagesToUse;
-        if (highscorePercentages.isEmpty()) {
-            highscorePercentagesToUse = DEFAULT_DISTRIBUTION_PERCENTAGES;
-        } else {
-            highscorePercentagesToUse = highscorePercentages.get();
-        }
-        if (highscorePercentagesToUse.size() < 2) {
-            throw new IllegalArgumentException("high score percentages must have at least 2 elements");
-        }
-        if (highscorePercentagesToUse.get(0) != 0) {
-            throw new IllegalArgumentException("high score percentages must start with 0");
-        }
-        if (highscorePercentagesToUse.get(highscorePercentagesToUse.size() - 1) != 100) {
-            throw new IllegalArgumentException("high score percentages must end with 100");
-        }
+    public List<MinigameScoreHit> getPlayerHighscoreDistributions(final UUID minigameTaskId) {
         final List<PlayerTaskStatistic> playerTaskStatistics = playerTaskStatisticRepository.findByMinigameTaskIdOrderByHighscore(
             minigameTaskId
         );
-        final List<MinigameHighscoreDistribution> highscoreDistributions = new ArrayList<>();
-        for (int i = 0; i < highscorePercentagesToUse.size() - 1; i++) {
-            final MinigameHighscoreDistribution highscoreDistribution = new MinigameHighscoreDistribution();
-            highscoreDistribution.setFromPercentage(highscorePercentagesToUse.get(i));
-            highscoreDistribution.setToPercentage(highscorePercentagesToUse.get(i + 1));
-            highscoreDistributions.add(highscoreDistribution);
-        }
+        final List<MinigameScoreHit> highscoreDistributions = new ArrayList<>();
+        highscoreDistributions.add(new MinigameScoreHit(0, 0));
+        highscoreDistributions.add(new MinigameScoreHit(100, 0));
```

Comment:
```suggestion
```
↑ This is a separate suggestion as GitHub didn't allow me to combine the one below with this one.